src/desktop/apps/article/components/__tests__/Article.test.js

```diff
@@ -179,3 +180,96 @@ describe('<Article />', () => {
     rendered.state().following.length.should.exist
   })
 })
+
+describe('<Article /> without infinite scroll', () => {
+  before(done => {
+    benv.setup(() => {
+      benv.expose({ $: benv.require('jquery'), jQuery: benv.require('jquery') })
+      sd.APP_URL = 'http://artsy.net'
+      sd.CURRENT_PATH =
+        '/article/artsy-editorial-surprising-reason-men-women-selfies-differently'
+      sd.CURRENT_USER = { id: '123' }
+      sd.ARTICLE_INFINITE_SCROLL = 'experiment'
+      done()
+    })
+  })
+
+  after(() => {
+    benv.teardown()
+  })
+
+  window.matchMedia = () => {
+    return {
+      matches: false,
+      addListener: () => {},
+      removeListener: () => {},
+    }
+  }
+
+  const rewire = require('rewire')('../layouts/Article')
+  const ArticleLayout = rewire.default
+  const InfiniteScrollArticle = require('desktop/apps/article/components/InfiniteScrollArticle')
+    .default
+  const { Article } = require('reaction/Components/Publishing')
+  const SuperArticleView = sinon.stub()
+  rewire.__set__('SuperArticleView', SuperArticleView)
+  const EditorialSignupView = sinon.stub()
+  rewire.__set__('EditorialSignupView', EditorialSignupView)
+
+  const getWrapper = props => {
+    return mount(
+      <ContextProvider currentUser={null}>
+        <ArticleLayout {...props} />
+      </ContextProvider>
+    )
+  }
+
+  let props
+  beforeEach(() => {
+    props = {
+      article: _.extend({}, fixtures.article, {
+        layout: 'standard',
+        vertical: {
+          name: 'Art Market',
+        },
+        published_at: '2017-05-19T13:09:18.567Z',
+        contributing_authors: [{ name: 'Kana' }],
+      }),
+      isSuper: false,
+      templates: {},
+    }
+  })
+
+  it('renders a standard article without infinite scroll', () => {
+    const rendered = getWrapper(props)
+
+    rendered.find(InfiniteScrollArticle).length.should.equal(0)
+    rendered.html().should.containEql('StandardLayout')
+  })
+
+  it('renders a standard article with ads', () => {
+    props.article = _.extend({}, props.article, {
+      display: {
+        name: 'Campaign 1',
+        panel: {
+          assets: [{ url: 'http://url.jpg' }],
+          headline: 'Ad Headline',
+          link: { url: 'http://link' },
+        },
+        canvas: {
+          assets: [{ url: 'http://url.jpg' }],
+          headline: 'Ad Headline',
+          link: { url: 'http://link' },
+          layout: 'standard',
+        },
+      },
+    })
+    const rendered = getWrapper(props)
+
+    const html = rendered.html()
+    html.should.containEql('DisplayPanel')
+    html.should.containEql('class="Canvas')
```

Comment:
We are moving away from `className`,  searching for the presence of components is  more future-proof if we import and search for the component directly:
`expect(rendered.find(DisplayPanel)).toHaveLength(0)`